src/R2Architecture.cpp

```diff
@@ -89,6 +92,11 @@ std::string SleighIdFromCore(RCore *core)
 		be = true;
 	if (!arch_it->second.compare("AARCH64"))
 		flavor = string("v8A");
+	// XXX: Since r2 only has a single sh mapping, this can get tricky with the current structure
+	if (!arch_it->second.compare("SuperH4"))
+		bits = 32; // XXX: SuperH4:LE:32
```

Comment:
`sh` in r2 only allows 32 bits so I don't think this is necessary. Also you already know that it's 32 bit because only `sh32` maps to `SuperH4`.